app/javascript/components/lease_ups/lottery_results_page/utils/processLotteryBuckets.js

```diff
@@ -0,0 +1,157 @@
+import { Preferences } from './preferences'
+
+const emptyBuckets = {
+  COP: [],
+  'V-COP': [],
+  DTHP: [],
+  'V-DTHP': [],
+  NRHP: [],
+  'V-NRHP': [],
+  L_W: [],
+  'V-L_W': [],
+  generalLottery: []
+}
+
+export const groupBuckets = (applicationPreferences) => {
+  return Object.values(applicationPreferences).reduce((acc, appPref) => {
+    const cleanApp = {
+      lottery_number:
+        appPref.application.lottery_number_manual ?? appPref.application.lottery_number,
+      unsorted_lottery_rank: appPref.application.unsorted_lottery_rank
+    }
+    if (appPref.application.general_lottery) {
+      acc.generalLottery.push(cleanApp)
+    } else {
+      if (acc[appPref.custom_preference_type]) {
+        acc[appPref.custom_preference_type].push(cleanApp)
+      }
+    }
+    return acc
+  }, emptyBuckets)
+}
+
+const unfilteredBucket = {
```

Comment:
suggestion: Move this variable declaration into `processUnfilteredBucket()`, since it's only used in that function.